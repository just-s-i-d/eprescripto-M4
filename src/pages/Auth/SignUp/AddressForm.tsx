import { Form, Input } from "antd";

const AddressForm = () => {
  return (
    <>
      <Form.Item
        label="Country"
        name="country"
        required
        rules={[
          {
            required: true,
            message: "Field cannot be empty",
          },
          {
            pattern: /^[A-Za-z]+$/,
            message: "Number and special characters not allowed",
          },
        ]}
      >
        <Input />
      </Form.Item>
      <Form.Item
        label="State"
        name="state"
        required
        rules={[
          {
            required: true,
            message: "Field cannot be empty",
          },
          {
            pattern: /^[A-Za-z]+$/,
            message: "Number and special characters not allowed",
          },
        ]}
      >
        <Input />
      </Form.Item>
      <Form.Item
        label="City"
        name="city"
        required
        rules={[
          {
            required: true,
            message: "Field cannot be empty",
          },
          {
            pattern: /^[A-Za-z]+$/,
            message: "Number and special characters not allowed",
          },
        ]}
      >
        <Input />
      </Form.Item>
      {/* <Form.Item
          label="Pincode"
          name="pincode"
          required
          rules={[
            {
              required: true,
              message: "Field cannot be empty",
            },
            {
              len: 6,
              message: "Enter a valid pincode",
            },
          ]}
        >
          <Input />
        </Form.Item> */}
    </>
  );
};

export default AddressForm;
