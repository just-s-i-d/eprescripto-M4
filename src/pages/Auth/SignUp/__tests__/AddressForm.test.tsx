import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import AddressForm from "../AddressForm";
import "@test/matchMedia";

describe("Address form test", () => {
  it("Render the Address fields correctly", () => {
    const { asFragment } = render(<AddressForm />);
    expect(screen.getByLabelText("Country")).toBeInTheDocument();
    expect(screen.getByLabelText("State")).toBeInTheDocument();
    expect(screen.getByLabelText("City")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
