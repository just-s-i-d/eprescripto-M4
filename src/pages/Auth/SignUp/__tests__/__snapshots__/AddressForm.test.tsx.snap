// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address form test Render the Address fields correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-form-item css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
      >
        <label
          class="ant-form-item-required"
          for="country"
          title="Country"
        >
          Country
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              aria-required="true"
              class="ant-input css-dev-only-do-not-override-1adbn6x"
              id="country"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-form-item css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
      >
        <label
          class="ant-form-item-required"
          for="state"
          title="State"
        >
          State
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              aria-required="true"
              class="ant-input css-dev-only-do-not-override-1adbn6x"
              id="state"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-form-item css-dev-only-do-not-override-1adbn6x"
  >
    <div
      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
    >
      <div
        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
      >
        <label
          class="ant-form-item-required"
          for="city"
          title="City"
        >
          City
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              aria-required="true"
              class="ant-input css-dev-only-do-not-override-1adbn6x"
              id="city"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
